### PR TITLE
First draft at a basic Text Repository

### PR DESCRIPTION
**Pull request by knewter**
This is in no way ready for merge, but wanted to begin a discussion around it.  It's a very simplistic idea, and includes the concept of a pluggable Strategy (though I'm not sure I love it being a mixin, pretty sure I don't...)

Things I dislike presently: Domain Objects have to know about their IDs.  At some level, this is a thing I can't see a way around.  Otherwise updating an object in a repository is ~impossible, and disambiguating two Domain Objects is impossible.  Think this is bog standard stuff, just sad a little.  Mayhaps it makes sense to have a RepositoryDecorator on domain objects that carries its ID information around, so that you can have a Normal DomainObject but it can be persisted, carry its persistence info with it, etc?  This would also make save easy - if it's wrapped, it's an existing thing.  If it's a raw domain object, you know to save it fresh in the persistence layer.

Thoughts?

### CODE REVIEW

File: core/spec/entities/content/text_spec.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ module Content
       describe Text do
         let(:text) { rand(30).times.map{|i| ('A'..'Z').to_a.sample}.join }
 
-        subject { Text.new(text) }
+        subject { Text.new(nil, text) }
```
Comment (knewter) [inline core/spec/entities/content/text_spec.rb:10]: This, specifically, gives me gobs of heartburn.  Having a RepositoryDecorator around these objects would make this unnecessary.

**Comment by knewter** [comment]
Given the idea that Persistence should be a side effect, it makes tons more sense to have that decorator.  Would love to have a better name for it.  Is this a pattern anyone's seen in the wild before?  Wonder what it's called elsewhere.

**Comment by parndt** [comment]
I very seriously favour the idea of persistence decorating the object with an identifier. VERY SERIOUSLY FAVOUR.

File: core/spec/entities/content/text_spec.rb (hunk 1 above)
Comment (parndt) [inline core/spec/entities/content/text_spec.rb:10]: I've been trying to comment but I keep fainting

**Comment by robyurkowski** [comment]
This isn't nearly as awesome as #53.

**Comment by robyurkowski** [comment]
... though to be clear, it's still awesome.

**Pull request closed without merging**